e2e/cypress/integration/messaging/search_group_message_spec.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+describe('Messaging', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('should returns results in GMs', () => {
+        let user1;
+        let user2;
+
+        //# text from invoke method
+        cy.get('#moreDirectMessage').click();
+
+        //# text from invoke method
+        cy.getAllByTestId('username').eq(1).children('span').invoke('text').then((text) => {
+            //# getting username text
+            user1 = text.split(' ')[0];
+
+            cy.get('.more-modal__name').contains(text).click();
+        });
+
+        cy.getAllByTestId('username').eq(1).children('span').invoke('text').then((text) => {
+            //# getting username text and removing @
+            user2 = text.split(' ')[0].substring(1);
+
+            //# select user from modal
+            cy.get('.more-modal__name').contains(text).click();
+
+            //# save values
+            cy.get('#saveItems').click();
+
+            //# searching 'lucho' text and hitting enter
+            cy.get('#post_textbox').clear().type('lucho').type('{enter}');
+
+            //# writing "in:" text in search input
+            cy.get('#searchBox').type('in:');
```

Comment:
Everything up to this point can most likely be API calls to setup the data, and then we are only concerned with typing in the search box and asserting results. You'll have to most likely start by making a group channel (https://api.mattermost.com/#tag/channels%2Fpaths%2F~1channels~1group%2Fpost) and then having one of the users post to that channel.

Reply:
ok makes sense, I will create the channel using the API. thanks